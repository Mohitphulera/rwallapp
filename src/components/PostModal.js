import React, { Component } from 'react'
import { request } from '../services/request'

const initialState = {
    loading: false,
    error: false,
    content: ''
}

export default class PostModal extends Component {
    constructor(props) {
        super(props)
        this.state = initialState
    }

    render() {
        const { loading, error, content } = this.state
        const { close, hidden } = this.props

        const footerMessage = !error ? (
            <p className="help">You can use CTRL + Enter or CMD + Enter to submit, too!</p>
        ) : (
            <p className="help is-danger">{error}</p>
        )
        const loadingIcon = loading ? (
            <span className="icon">
                <i className="fa fa-circle-o-notch fa-spin" />
            </span>
        ) : null

        return (
            <div hidden={hidden}>
                <div className="modal is-active">
                    <div className="modal-background" onClick={() => close()} />
                    <div className="modal-card">
                        <header className="modal-card-head">
                            <p className="modal-card-title">Make a Post</p>
                            <button className="delete" onClick={() => close()} aria-label="close" />
                        </header>
                        <section className="modal-card-body">
                            <form>
                                <div className="field">
                                    <div className="control has-icons-left has-icons-right">
                                        <textarea
                                            value={content}
                                            onChange={({ target: { value } }) => this.setState({ content: value })}
                                            onKeyUp={({ ctrlKey, key }) => (ctrlKey && key === 'Enter' ? this.submit() : null)}
                                            name="message"
                                            className="textarea"
                                            type="text"
                                            required
                                        />
                                    </div>
                                </div>
                            </form>
                        </section>
                        <footer className="modal-card-foot">
                            <button onClick={() => this.submit()} className="button is-link">
                                <span>Submit</span>
                                {loadingIcon}
                            </button>
                            <button className="button" onClick={() => close()}>
                                Cancel
                            </button>
                            {footerMessage}
                        </footer>
                    </div>
                </div>
            </div>
        )
    }

    submit() {
        this.setState({
            loading: true,
            error: ''
        })
        request
            .post('posts/', { message: this.state.content })
            .then(() => {
                this.props.postCreated()
                this.close(false)
            })
            .catch(response => {
                let error = 'Failed to submit post.'
                if (response && response.message) {
                    error = response.message.join(' ')
                }
                this.setState({
                    loading: false,
                    error
                })
            })
    }

    reset() {
        this.setState(initialState)
    }

    close(preserve = true) {
        if (!preserve) {
            this.reset()
        }
        this.props.close()
    }
}
