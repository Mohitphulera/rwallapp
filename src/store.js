import { createStore } from 'redux'

const initialState = {
    user: null
}

const SET_USER = 'SET_USER'

const reducer = (state = initialState, action) => {
    switch(action.type) {
        case SET_USER: {
            console.log(action.payload ? 'logging in' : 'logging out')
            return {
                user: action.payload
            }
        }
        default: {
            return state
        }
    }
}


export function setUser(user = null) {
    return {
        type: SET_USER,
        payload: user
    }
}

export const store = createStore(reducer)