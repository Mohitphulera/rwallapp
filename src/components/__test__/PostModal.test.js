import React from 'react'
import ReactDom from 'react-dom'
import PostModal from './../PostModal'
import {isTSAnyKeyword} from  '@babel/types'
import { Provider } from 'react-redux'
import { store } from './../../store'
import renderer from "react-test-renderer"

it("renders without crashing",()=> {
    const div = document.createElement("div");
    ReactDom.render( <Provider store={store}>
        <PostModal />
      </Provider>,div);

})

it("matches snapshot & renders correctly",() => {
    const tree=renderer.create(<Provider store={store}>
        <PostModal />
      </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})