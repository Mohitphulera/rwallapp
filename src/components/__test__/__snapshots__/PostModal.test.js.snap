// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot & renders correctly 1`] = `
<div>
  <div
    className="modal is-active"
  >
    <div
      className="modal-background"
      onClick={[Function]}
    />
    <div
      className="modal-card"
    >
      <header
        className="modal-card-head"
      >
        <p
          className="modal-card-title"
        >
          Make a Post
        </p>
        <button
          aria-label="close"
          className="delete"
          onClick={[Function]}
        />
      </header>
      <section
        className="modal-card-body"
      >
        <form>
          <div
            className="field"
          >
            <div
              className="control has-icons-left has-icons-right"
            >
              <textarea
                className="textarea"
                name="message"
                onChange={[Function]}
                onKeyUp={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </form>
      </section>
      <footer
        className="modal-card-foot"
      >
        <button
          className="button is-link"
          onClick={[Function]}
        >
          <span>
            Submit
          </span>
        </button>
        <button
          className="button"
          onClick={[Function]}
        >
          Cancel
        </button>
        <p
          className="help"
        >
          You can use CTRL + Enter or CMD + Enter to submit, too!
        </p>
      </footer>
    </div>
  </div>
</div>
`;
